src/js/ripple/orderbook.js

```diff
@@ -1176,4 +1335,307 @@ OrderBook.prototype.is_valid = function() {
+      autobridgedTakerPays = this.getOfferTakerPaysFunded(legOneOffer);
+
+      // Update funded amount since leg two offer was not completely consumed
+      legTwoOffer.taker_gets_funded = this.getOfferTakerGetsFunded(legTwoOffer)
+        .subtract(autobridgedTakerGets)
+        .to_text();
+      legTwoOffer.taker_pays_funded = legTwoTakerPaysFunded
+        .subtract(legOneTakerGetsFunded)
+        .to_text();
+
+      legOnePointer++;
+    } else {
+      // Rare case when leg one out and leg two in are equal. Neither leg needs
+      // to be clamped and both offers are fully consumed
+      autobridgedTakerGets = this.getOfferTakerGetsFunded(legTwoOffer);
+      autobridgedTakerPays = this.getOfferTakerPaysFunded(legOneOffer);
+
+      legOnePointer++;
+      legTwoPointer++;
+    }
+
+    var autobridgedOffer = this.formatAutobridgedOffer(
+      autobridgedTakerGets,
+      autobridgedTakerPays
+    );
+    autobridgedOffer.index = legOnePointer + legTwoPointer;
+
+    this._offersAutobridged.push(autobridgedOffer);
+  }
+};
+
+/**
+ * Format an autobridged offer and compute synthetic values (e.g. quality)
+ *
+ * @param [Amount] takerGets
+ * @param [Amount] takerPays
+ *
+ * @return [Object]
+ */
+
+OrderBook.prototype.formatAutobridgedOffer = function(takerGets, takerPays) {
+  assert(takerGets instanceof Amount, 'Autobridged taker gets is invalid');
+  assert(takerPays instanceof Amount, 'Autobridged taker pays is invalid');
+
+  var autobridgedOffer = {};
+  var quality = takerPays.divide(takerGets);
+
+  autobridgedOffer.TakerGets = {
+    value: takerGets.to_text(),
+    currency: this._currencyGets.to_hex(),
+    issuer: this._issuerGets
+  };
+
+  autobridgedOffer.TakerPays = {
+    value: takerPays.to_text(),
+    currency: this._currencyPays.to_hex(),
+    issuer: this._issuerPays
+  };
+
+  autobridgedOffer.quality = quality.to_text();
+
+  autobridgedOffer.taker_gets_funded = autobridgedOffer.TakerGets.value;
+  autobridgedOffer.taker_pays_funded = autobridgedOffer.TakerPays.value;
+
+  autobridgedOffer.autobridged = true;
+
+  autobridgedOffer.BookDirectory = this.convertOfferQualityToHex(quality);
+
+  return autobridgedOffer;
+};
+
+/**
+ * Formats an offer quality amount to a hex that can be parsed by
+ * Amount.parse_quality
+ *
+ * @param [Amount] quality
+ *
+ * @return [String]
+ */
+
+OrderBook.prototype.convertOfferQualityToHex = function(quality) {
+  assert(quality instanceof Amount, 'Quality is not an amount');
+
+  var so = new SerializedObject();
+  Types.Quality.serialize(so, quality.to_text() + OrderBook.IOU_SUFFIX);
+
+  return so.to_hex();
+};
+
+/**
+ * Remove funds clamp on leg one offer. This is necessary when the two offers
+ * are owned by the same account. In this case, it doesn't matter if offer one
+ * is not fully funded. Leg one out goes to leg two in and since its the same
+ * account, an infinite amount can flow.
+ *
+ * @param [Object] legOneOffer - IOU:XRP offer
```

Comment:
I got the impression we have JSDoc linting turned on: http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values

[bracket-syntax] denotes optionality?